src/lib/shadow-shim-helper-rs/src/rootedcell/rc.rs

```diff
@@ -92,23 +102,27 @@ impl<T> RootedRc<T> {
     /// Otherwise the underlying reference count will simply not be decremented,
     /// ultimately resulting in the enclosed value never being dropped.
     pub fn safely_drop(mut self, root: &Root) {
-        assert_eq!(
-            root.tag, self.tag,
-            "Tried using a root {:?} instead of {:?}",
-            root.tag, self.tag
-        );
-        let internal = self.internal.take().unwrap();
-        let drop_internal = {
-            // SAFETY: pointer points to valid data by construction.
-            let internal = unsafe { internal.as_ref() };
-            internal.dec_strong();
-            internal.strong_count.get() == 0
-        };
-        if drop_internal {
-            // SAFETY: There are no remaining strong references to
-            // self.internal, and we know that no other threads could be
-            // manipulating the reference count in parallel since we have the
-            // root lock.
+        let internal: &RootedRcInternal<T> = self.validate_thread_exclusive_access(root);
```

Comment:
Drop the `&RootedRcInternal<T>` before freeing the memory like done in `RootedRcWeak`?

Reply:
No need; removed it in the other one.